### ISSUE TITLE
Releases and Compatibility

### ISSUE THREAD

**Issue by kitswas** [hust-diangroup/ns3-ai#112]
The [Latest Release](https://github.com/hust-diangroup/ns3-ai/releases/latest) is v1.2.0 which is aligned with ns-3.38. It was released almost a year earlier in June 2023.  
v1.2.0 can be built with examples enabled on ns-3.40.

I was also able to build the latest commit off main (7dc5b1b8d574eb587b088fa959a230c2b4e1aa75) which is supposed to align with ns-3.41.

Builds off `main`, however, fail when examples are enabled and the ns3 version is 3.40.
Error occurs at [examples/multi-bss/vr-app/model/vr-burst-generator.cc#L64](https://github.com/hust-diangroup/ns3-ai/blob/7dc5b1b8d574eb587b088fa959a230c2b4e1aa75/examples/multi-bss/vr-app/model/vr-burst-generator.cc#L64)
> no matching function for call to ‘MakeEnumAccessor<ns3::VrBurstGenerator::VrAppName>(ns3::VrBurstGenerator::VrAppName ns3::VrBurstGenerator::*)’

Please tell which commit supports ns3.40 with examples enabled.

**Comment by lemonixxxx** [comment]
I meet the same problem if you slove the problem please tell me thank very much!!!!!(besides I'm not a native speaker if any word offenses you please forgive me thank's a lot)

**Comment by kitswas** [comment]
I sidestepped the problem by excluding the offending example(s) from the CMake build target 'all'.

You can find this hack on my fork at [`commit 66f5228c75cbc5c9542a34475c04a715014004ac`](https://github.com/kitswas/ns3-ai/commit/66f5228c75cbc5c9542a34475c04a715014004ac).

Remember to reconfigure and rebuild after making the change.

**Comment by SirCrocker** [comment]
Hi, I encountered the same problem and was able to circumvent it by commenting the line "add_subdirectory(multi-bss)" inside the examples' [CMakeLists.txt](../blob/main/examples/CMakeLists.txt). A better solution is updating the MakeEnumAccessor of the class VrBurstGenerator, I did not try to do it though.

**Comment by Muhammaduazir69** [comment]
![ns3-installation-ns-3 42](https://github.com/user-attachments/assets/c6ca4655-c1e9-4137-a84b-27ebd0910fab)

Succesfully configured NS3-AI with NS-3.42(Nov 23, 2024)

Docker Envirenment information ! conda

Ubuntu 20.04
protobuf tag v21.12 with cmake installation
NS3-AI/CmakeLists.txt file modifications
Replace line 64 at vr-burst-generation.cc file MakeEnumAccessor<VrAppName>(&VrBurstGenerator::m_appName),

Done with dynamic resources allocation in 5G-lena(NR) by implementing SDN based Network Slicing using NS3-AI in NS3

**Comment by Felicialalala** [comment]
> ![ns3-ai-installation2](https://github.com/user-attachments/assets/0fa6154f-6b0d-41f3-b265-d0109ee25105)
> 
> Succesfully configured NS3-AI with NS-3.42(Nov 23, 2024)
> 
> Docker Envirenment information ! conda
> 
> Ubuntu 20.04 protobuf tag v21.12 with cmake installation NS3-AI/CmakeLists.txt file modifications Replace line 64 at vr-burst-generation.cc file MakeEnumAccessor(&VrBurstGenerator::m_appName),
> 
> Done with dynamic resources allocation in 5G-lena(NR) by implementing SDN based Network Slicing using NS3-AI in NS3

@Muhammaduazir69 Hi，I'm currently trying to run ns3-ai on ns3.42, but encountered some issues. When running the a+b example, I got a ModuleNotFoundError, which seems like a Python binding problem. Did you experience similar issues before? Could you please share your modifications to the CMakeList file?

**Comment by Muhammaduazir69** [comment]
@Felicialalala Yes i faced the pybind11 & protobuf errors like you facing 'modulenotfounderror', i apply custom changes to fix these all errors, Yes i can share my Cmakelist & whole VM that's configuring NS3, NS3-AI, NR(5g-lena)

**Comment by Felicialalala** [comment]
Hi，@Muhammaduazir69 I downgraded the Python version to 3.10, which solved the problem with pybind. However, I encountered some other issues. If it's convenient for you, please share your code repository with me. I would be extremely grateful! (Currently, I'm trying to use reinforcement learning to make some attempts at intelligent scheduling in the MAC layer within ns3-ai and 5g-lena.)

**Comment by Picxs** [comment]
Hi, @Felicialalala and @Muhammaduazir69, how's your research going? If you don't mind, I'd like to ask for some advice. I'm going to try to build a DRL environment using ns3-ai and 5G Lena for a project and my college thesis. I haven't started yet, and I've seen many comments on the difficulty of integrating everything in the latest versions of ns-3. I should use an earlier version?. I've seen recommendations about 3.38, which ns3-ai has native support for, but I really don't know. If you wouldn't mind sharing your repository, I'd be extremely grateful for some reference on how to do it.
